Assignment2/Q1/src/BinaryTree.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Created by Alice Colt.
+ *
+ * Implementaiton of a Binary Tree with functions for insert, serach and takeAncestors
+ *
+ */
+public class BinaryTree {
+    private NodeBinaryTree root;
+
+    //Constructor
+    public BinaryTree(NodeBinaryTree new_root){
+        root = new_root;
+    }
+
+
+    public void insert(NodeBinaryTree new_node) {
+
+        NodeBinaryTree actualNode = root;
+
+        if (new_node != null) {
+
+            while (actualNode != null) {
+                if (new_node.compareTo(actualNode) == 1 && actualNode.getLeft() == null) {
+                    actualNode.setLeft(new_node);
+                    new_node.setParent(actualNode);
+                    break;
+                }//if
+                else if (new_node.compareTo(actualNode) == -1 && actualNode.getRight() == null) {
+                    actualNode.setRight(new_node);
+                    new_node.setParent(actualNode);
+                    break;
+                }//else if
+                else if (new_node.compareTo(actualNode) == 0) {
+                    System.out.println(" The node added is a duplicate ");
+                    break;
+                }//else if
+                else {
+                    if (new_node.compareTo(actualNode) == 1) {
+                        actualNode = actualNode.getLeft();
+                    }//if
+                    else if(new_node.compareTo(actualNode) == -1) {
+                        actualNode = actualNode.getRight();
+                    }//else if
+                }//else
+            }//while
+        }//if
+    }//insert
+
+
+    // Get ancestors
+    public void takeAncestors(int value)
+    {
+        NodeBinaryTree actualNode = root;
+
+        // Check if child is null
+        if (actualNode == null) {
+            System.out.println("Node null");
+        }//if
+
+        // Search Call
+        actualNode = search(root, value);
```

Comment:
Search returns null if the node isn't found. One alternative is for search to throw an exception if the item is not found. However, I'd expect a function that prints out a list of ancestors of a node wouldn't just silently print nothing if the node is not present. If nothing else, the output is indistinguishable from the output we'd expect for the root node (which has no ancestors).

Therefore, I'd suggest throwing an exception if actualNode is null.